pennylane/ops/qubit/hamiltonian.py

```diff
@@ -567,7 +567,7 @@ def __mul__(self, a):
         r"""The scalar multiplication operation between a scalar and a Hamiltonian."""
         if isinstance(a, (int, float)):
             self_coeffs = copy(self.coeffs)
-            coeffs = qml.math.multiply(qml.math.cast_like([a], self_coeffs), self_coeffs)
+            coeffs = a * self_coeffs
```

Comment:
Hi @puzzleshark, how does this solution work with interfaces (TF, Torch, JAX)?

Reply:
there is a check that a must be a int or float, so I don't think there can be a problem

Reply:
changed to `qml.math.multiply` @antalszava